src/main/java/use_case/search/SearchInputBoundary.java

```diff
@@ -0,0 +1,18 @@
+package use_case.search;
+
+import data_access.movies.MovieDBDataAccessException;
+
+/**
+ * Input boundary for the search by criteria use case.
+ */
+public interface SearchInputBoundary {
+
+    /**
+     * Execute the search by criteria use case.
+     *
+     * @param inputData the search criteria input data.
+     * @return the search results (output data).
+     * @throws MovieDBDataAccessException if there is an error during the search process.
+     */
+    SearchByCriteriaOutputData execute(SearchByCriteriaInputData inputData) throws MovieDBDataAccessException;
```

Comment:
The input boundary shouldn't return anything to fit the flow of control of clean architecture. As mentioend in a previous comment, the interactor implementing the input boundary should be the one to call the output boundary's methods with the output data as the argument.